File: Sort&Searching/binarysearch.cpp
#include <iostream>

using namespace std;

int binarySearch(int arr[],int l,int r,int x)
{
	int mid = l+(r-l)/2;

	if (arr[mid] == x)
		return mid;
	else if (arr[mid] < x)
		return binarySearch(arr,mid+1,r,x);
	else if (arr[mid] > x)
		return binarySearch(arr,l,mid-1,x);
	else
		return -1;
}

int main()
{
	int arr[] = {1,2,3,4,5};
	int len = sizeof(arr)/sizeof(arr[0]);
	int result = binarySearch(arr,0,len-1,5);
	(result != -1) ? cout<<"Is the "<<result+1<<" element of the array"<<endl : cout<<"Not found"<<endl;
	return 0;
}

File: Java/HelloWorld/BinarySearch.java
class BinarySearch
{
	public static boolean bs(int[] arr,int lo,int hi,int b)
	{
		int mid = (lo+hi)/2;
		if (lo==hi) return false;
		else if (arr[mid] == b) return true;
		else if (arr[mid] < b) return bs(arr,mid+1,hi,b);
		else if (arr[mid] > b) return bs(arr,lo,mid-1,b);
		else return false;
	}

	public static void main(String[] args) {
		int arr[] = {1,2,3,4,5};

		int lo=0;
		int hi=arr.length-1;

		String result = bs(arr,lo,hi,10) ? "Yes" : "No";
		System.out.println("Found : "+result);
	}
}